### PR TITLE
Migrate android views to devicelab.

### PR DESCRIPTION
**Pull request by godofredoc**
Migrate android view out of recipes.

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by godofredoc** [comment]
Hi Reid, as part of simplifying testing we are deprecating https://cs.opensource.google/flutter/recipes/+/main:recipes/flutter/android_views.py and migrating the test to a regular devicelab test using the new emulator context. Please take a look at this pr.

Also this is how the generic recipe with this test looks like: https://chromium-swarm.appspot.com/task?id=675600a99df8c110

File: dev/devicelab/lib/tasks/android_views_test.dart [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        );
+
+        // Any gradle command downloads gradle if not already present in the cache.
+        // ./gradlew dependencies downloads any gradle defined dependencies to the cache.
+        // https://docs.gradle.org/current/userguide/viewing_debugging_dependencies.html
+        // Downloading gradle and downloading dependencies are a common source of flakes
+        // and moving those to an infra step that can be retried shifts the blame
+        // individual tests to the infra itself.
+        /////// android_path = views_test_dir.join('android')
+        /////// with api.context(env=env, env_prefixes=env_prefixes, cwd=android_path):
```
Comment (keyonghan) [inline dev/devicelab/lib/tasks/android_views_test.dart:35]: Remove.

File: dev/devicelab/lib/tasks/android_views_test.dart [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+          environment: environment,
+          workingDirectory: '${flutterDirectory.path}/dev/integration_tests/android_views'
+        );
+
+        // Any gradle command downloads gradle if not already present in the cache.
+        // ./gradlew dependencies downloads any gradle defined dependencies to the cache.
+        // https://docs.gradle.org/current/userguide/viewing_debugging_dependencies.html
+        // Downloading gradle and downloading dependencies are a common source of flakes
+        // and moving those to an infra step that can be retried shifts the blame
+        // individual tests to the infra itself.
```
Comment (keyonghan) [inline dev/devicelab/lib/tasks/android_views_test.dart:33]: Nit: reformat the doc.

File: dev/devicelab/lib/tasks/android_views_test.dart [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+import '../framework/utils.dart';
+
+
+/// Tests the following Android lifecycles: Activity#onStop(), Activity#onResume(), Activity#onPause(),
+/// and Activity#onDestroy() from Dart perspective in debug, profile, and release modes.
+TaskFunction androidViewsTest({
+  Map<String, String>? environment,
+}){
+    return () async {
+        //return TaskResult.failure('aaa');
```
Comment (keyonghan) [inline dev/devicelab/lib/tasks/android_views_test.dart:16]: Nit: remove.

**Review by keyonghan — APPROVED** [review]
LGTM with nits.

File: dev/devicelab/lib/tasks/android_views_test.dart (hunk 1 above)
Comment (godofredoc) [reply dev/devicelab/lib/tasks/android_views_test.dart:35]: Done

**Review by yusuf-goog — APPROVED** [review]

File: dev/devicelab/lib/tasks/android_views_test.dart (hunk 3 above)
Comment (godofredoc) [reply dev/devicelab/lib/tasks/android_views_test.dart:16]: Done

File: dev/devicelab/lib/tasks/android_views_test.dart (hunk 2 above)
Comment (godofredoc) [reply dev/devicelab/lib/tasks/android_views_test.dart:33]: Done!

**Pull request merged**